### ISSUE TITLE
Angular: Empty master-detail list is not updating when replacing data array

### ISSUE THREAD

**Issue by ttntrifork** [eclipsesource/jsonforms#1667]
**Describe the bug**
Empty master-detail list is not updating itself when the data array is being replaced using `jsonFormsService.updateCore(Actions.update, ...)`. 
The master-detail list will only update when it is forced to redraw itself from browser window being resized or when adding another entry through the master-detail control itself.

See https://spectrum.chat/jsonforms/general/angular-empty-master-detail-list-not-updating~76ad2d65-d1df-426c-bf6e-00f00ea6c6eb

**To Reproduce**
1. Clone https://github.com/ttntrifork/jsonforms-angular-seed
2. ng serve
3. Delete all entries in the master-detail list using the UI (delete icon)
4. Click `Import data` button

**Expected behavior**
The master-detail list updates to show the content of the new data array

**Browser**
Chrome 87.0.4280.88

**Comment by ttntrifork** [comment]
I ran into another problem with the master-detail list:  
Calling `jsonFormsAngularService.setReadonly(true)` disables a master-detail component, but calling `jsonFormsAngularService.setReadonly(false)` afterwards only enables the master list controls, not the nested details controls. 

I have pushed a demonstration of this to the same fork.  
See https://github.com/ttntrifork/jsonforms-angular-seed/commit/52200399972801a5c79eb8e04dc2800b5630a3cb

**To Reproduce**
1. Clone https://github.com/ttntrifork/jsonforms-angular-seed
2. ng serve
3. Check the `Read Only` checkbox
4. Uncheck the `Read Only` checkbox

**Expected behavior**
The nested details controls should be re-enabled when unchecking the `Read Only` checkbox
